### ISSUE TITLE
Add Skip to Uniquify/Overwrite/Prompt

### ISSUE THREAD

**Issue by Sopor** [gyng/save-in#94]
It would be nice if the `Uniquify/Overwrite/Prompt` also have a Skip option.

**Comment by gyng** [comment]
@Sopor- can you elaborate? I don't think I quite understand what "skip" would do when there are conflicting filenames.

**Comment by Sopor-** [comment]
You already have `Show notification on failed download` and some other notification, so if a file i want to download already exists it should skip downloading it and only show a message `Skipping $FILENAME$ because it already exists.`.
If someone want to not show any message at all there could be an option to turn it off completely by adding the option `Show notification on skipped download`.

**Comment by gyng** [comment]
This will depend on browser support as extensions have no access to the file system. Unfortunately, the API does not seem to support a "skip", so this won't be possible until browsers implement it.

See: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/downloads/FilenameConflictAction

**Comment by Sopor-** [comment]
So it is not possible to detect when a file name exists and do something else? For example, don't open the save as window?

**Comment by gyng** [comment]
@Sopor- No, it's not possible at this moment. The web extension API simply does not give extensions access to the file system for security and privacy reasons. The best that can be done is "prompt", which will open a naming window on a conflicting filename, but that is only supported in Chrome.

I can keep a list of downloaded URLs to check against, but I don't think that will be very useful.

**Comment by gyng** [comment]
Tagging bugzilla issue https://bugzilla.mozilla.org/show_bug.cgi?id=1425515
